lib/bpf/common_test.go

```diff
@@ -17,145 +17,62 @@ limitations under the License.
-
-		switch os.Args[1] {
-		case reexecInCGroupCmd:
-			// Get the command to run passed as the 3rd argument.
-			cmd := os.Args[2]
-
-			err = waitAndRun(cmd)
-		case networkInCgroupCmd:
-			// Get the endpoint to call.
-			endpoint := os.Args[2]
-
-			err = callEndpoint(endpoint)
-		default:
-			os.Exit(2)
-		}
-
-		if err != nil {
-			// Something went wrong, exit with error.
-			os.Exit(1)
-		}
-
-		// The rexec was handled and nothing bad happened.
-		os.Exit(0)
-	}
-
-	os.Exit(m.Run())
-}
-
 // TestCheckAndSetDefaults makes sure defaults are set when the user does not
 // provide values for the page sizes and hard coded values (like zero or a
 // specific page size) are respected when given.
-func TestCheckAndSetDefaults(t *testing.T) {
-	var perfBufferPageCount = defaults.PerfBufferPageCount
-	var openPerfBufferPageCount = defaults.OpenPerfBufferPageCount
-	var zeroPageCount = 0
+func TestBPFConfig_CheckAndSetDefaults(t *testing.T) {
+	perfBufferPageCount := defaults.PerfBufferPageCount
+	openPerfBufferPageCount := defaults.OpenPerfBufferPageCount
+	zeroPageCount := 0
 
 	var tests = []struct {
-		inConfig  *servicecfg.BPFConfig
-		outConfig *servicecfg.BPFConfig
+		name string
+		got  *servicecfg.BPFConfig
+		want *servicecfg.BPFConfig
 	}{
-		// Empty values get defaults.
 		{
-			inConfig: &servicecfg.BPFConfig{
-				CommandBufferSize: nil,
-				DiskBufferSize:    nil,
-				NetworkBufferSize: nil,
-			},
-			outConfig: &servicecfg.BPFConfig{
+			name: "all defaults",
+			got:  &servicecfg.BPFConfig{},
+			want: &servicecfg.BPFConfig{
 				CommandBufferSize: &perfBufferPageCount,
 				DiskBufferSize:    &openPerfBufferPageCount,
 				NetworkBufferSize: &perfBufferPageCount,
+				CgroupPath:        defaults.CgroupPath,
 			},
 		},
-		// Values are not wiped out with defaults.
 		{
-			inConfig: &servicecfg.BPFConfig{
+			name: "values set",
+			got: &servicecfg.BPFConfig{
 				CommandBufferSize: &zeroPageCount,
 				DiskBufferSize:    &zeroPageCount,
 				NetworkBufferSize: &perfBufferPageCount,
+				CgroupPath:        "/my/cgroup/",
 			},
-			outConfig: &servicecfg.BPFConfig{
+			want: &servicecfg.BPFConfig{
 				CommandBufferSize: &zeroPageCount,
 				DiskBufferSize:    &zeroPageCount,
 				NetworkBufferSize: &perfBufferPageCount,
+				CgroupPath:        "/my/cgroup/",
 			},
 		},
 	}
-
-	for _, tt := range tests {
-		err := tt.inConfig.CheckAndSetDefaults()
-		require.NoError(t, err)
-		require.Equal(t, *tt.outConfig.CommandBufferSize, *tt.inConfig.CommandBufferSize)
-		require.Equal(t, *tt.outConfig.DiskBufferSize, *tt.inConfig.DiskBufferSize)
-		require.Equal(t, *tt.outConfig.NetworkBufferSize, *tt.inConfig.NetworkBufferSize)
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := test.got.CheckAndSetDefaults()
+			require.NoError(t, err, "CheckAndSetDefaults errored")
+
+			if diff := cmp.Diff(test.want, test.got); diff != "" {
+				t.Errorf("CheckAndSetDefaults mismatch (-want +got)\n%s", diff)
+			}
```

Comment:
Let's be consistent
```suggestion
			require.Empty(t, cmp.Diff(test.want, test.got), "CheckAndSetDefaults mismatch (-want +got)\n%s", diff)
			
```

Reply:
This just makes the error message worse to read, imo.